src/test/java/com/networknt/schema/regex/Issue814Test.java

```diff
@@ -8,40 +8,50 @@ class Issue814Test {
 
     @Test
     void jdkTypePattern() {
-        JDKRegularExpression ex = new JDKRegularExpression("^list|date|time|string|enum|int|double|long|boolean|number$");
+        JDKRegularExpression ex = new JDKRegularExpression("^(list|date|time|string|enum|int|double|long|boolean|number)$");
         assertTrue(ex.matches("list"));
         assertTrue(ex.matches("string"));
         assertTrue(ex.matches("boolean"));
         assertTrue(ex.matches("number"));
         assertTrue(ex.matches("enum"));
+        assertFalse(ex.matches("listZ"));
+        assertFalse(ex.matches("AenumZ"));
+        assertFalse(ex.matches("Anumber"));
     }
 
     @Test
     void jdkOptionsPattern() {
-        JDKRegularExpression ex = new JDKRegularExpression("^\\d*|[a-zA-Z_]+$");
+        JDKRegularExpression ex = new JDKRegularExpression("^\\d+|[a-zA-Z_]+$");
+        assertTrue(ex.matches("0external"));
         assertTrue(ex.matches("external"));
         assertTrue(ex.matches("external_gte"));
         assertTrue(ex.matches("force"));
         assertTrue(ex.matches("internal"));
+        assertFalse(ex.matches("internal9"));
     }
 
     @Test
     void joniTypePattern() {
-        JoniRegularExpression ex = new JoniRegularExpression("^list|date|time|string|enum|int|double|long|boolean|number$");
+        JoniRegularExpression ex = new JoniRegularExpression("^(list|date|time|string|enum|int|double|long|boolean|number)$");
         assertTrue(ex.matches("list"));
         assertTrue(ex.matches("string"));
         assertTrue(ex.matches("boolean"));
         assertTrue(ex.matches("number"));
         assertTrue(ex.matches("enum"));
+        assertFalse(ex.matches("listZ"));
+        assertFalse(ex.matches("AenumZ"));
+        assertFalse(ex.matches("Anumber"));
     }
 
     @Test
     void joniOptionsPattern() {
-        JoniRegularExpression ex = new JoniRegularExpression("^\\d*|[a-zA-Z_]+$");
+        JoniRegularExpression ex = new JoniRegularExpression("^\\d+|[a-zA-Z_]+$");
+        assertTrue(ex.matches("0external"));
         assertTrue(ex.matches("internal"));
         assertTrue(ex.matches("external"));
         assertTrue(ex.matches("external_gte"));
```

Comment:
I think that similar examples can be removed, and the next can be added:
(matches left part: starts with digit:)
assertTrue - "5"
assertTrue - "55"
assertTrue - "5%"
(matches right part: ends with alpha-or-undescore:)
assertTrue - "a"
assertTrue - "aa"
assertTrue - "%a"
assertTrue - "%_"
(matches both parts)
assertTrue - "55aa"
assertTrue - "5%%a"

assertFalse: ""
assertFalse: "%"
assertFalse: "a5"

Reply:
Sure, lemme replace them.